util/db/repository_secrets_test.go

```diff
@@ -85,71 +81,55 @@ func TestSecretsRepositoryBackend_CreateRepository(t *testing.T) {
 	t.Run("will return proper error if secret does not have expected label", func(t *testing.T) {
 		// given
 		t.Parallel()
-		secret := &corev1.Secret{}
-		s := secretsRepositoryBackend{}
-		updatedSecret := s.repositoryToSecret(repo, secret)
-		delete(updatedSecret.Labels, common.LabelKeySecretType)
-		f := setupWithK8sObjects(updatedSecret)
```

Comment:
Manually crafting a secret missing the proper label and then adding it to the fake client directly is problematic, because the bespoke secret object doesn't have the name or namespace field set. 

I suspect this is why the custom reactors were necessary: fake client didn't return `AlreadyExists`, because the object originally added didn't have metadata to conflict with the new object.

Instead of manually crafting the secret, I've modified the test to just use `CreateRepository` via the standard library call and then update the Secret directly to remove the label.